src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,112 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int GROW_FACTOR = 2;
+
+    private Node<K,V>[] nodes;
+    private int size;
+
+    @SuppressWarnings("unchecked")
+    public MyHashMap() {
+        nodes = (Node<K, V>[]) new Node[INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        size += putNode(nodes, new Node<>(key, value), false);
+        resize();
     }
 
     @Override
     public V getValue(K key) {
+        int bucket = getBucket(nodes, key);
+        Node<K, V> node = nodes[bucket];
+
+        while (node != null) {
+            if (keysEqual(node.key, key)) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    @SuppressWarnings("unchecked")
+    private void resize() {
+        if (size >= nodes.length * LOAD_FACTOR) {
+            int newCapacity = nodes.length * GROW_FACTOR;
+            Node<K, V>[] newNodes = (Node<K, V>[]) new Node[newCapacity];
+            for (int i = 0; i < nodes.length; i++) {
+                Node<K, V> node = nodes[i];
+                if (node == null) {
+                    continue;
+                }
+                do {
+                    putNode(newNodes, node, true);
```

Comment:
You can reuse method put of your map

Reply:
Not sure what you mean, sorry.
This method (resize()) is called from the put() method.,,
So...